packages/react-router-dev/vite/plugin.ts

```diff
@@ -2510,21 +2510,21 @@ async function prerenderResourceRoute(
     .replace(/\/$/g, "");
   let request = new Request(`http://localhost${normalizedPath}`, requestInit);
   let response = await handler(request);
-  let text = await response.text();
+  let content = Buffer.from(await response.arrayBuffer());
```

Comment:
Naïve question, but should we be handling the response's `content-type` before converting to a Buffer? Saving directly from an ArrayBuffer without checking could lead to incorrect file encoding.

Would it make sense to ensure the correct format based on content-type before writing to disk?

This sorta thing:
![image](https://github.com/user-attachments/assets/d5d04ddc-8157-452c-ab58-18d8d6d554f8)

Reply:
My understanding is a buffer simply holds the content bytes. It doesn't track encoding at all.

It seems most correct to just pass the response bytes directly into the file without even trying to interpret the bytes (meaning we cannot possibly misinterpret them)

Reply:
Nice! This is a great solution in that case.